File: easy/strings/python/125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        s = "".join(c.lower() for c in s if c.isalnum())
        for i in range(len(s)//2):
            back = len(s) - i - 1
            if s[i] != s[back]:
                return False
        return True

        # Extra pythonic:
        return s == s[::-1]


print(Solution().isPalindrome("A man, a plan, a canal: Panama"))
print(Solution().isPalindrome("racecar"))
print(Solution().isPalindrome("avani"))

File: easy/strings/cpp/125-valid-palindrome.cpp
#include "../../../stdc++.h"

using std::string;
using std::vector;
class Solution
{
public:
    bool isPalindrome(string s)
    {
        vector<char> new_str;
        for (char c : s)
        {
            if (isalnum(c))
            {
                c = tolower(c);
                new_str.push_back(c);
            }
        }
        for (size_t i = 0; i < new_str.size() / 2; i++)
        {
            int back = new_str.size() - i - 1;
            if (new_str[i] != new_str[back])
            {
                return false;
            }
        }
        return true;
    }
};

int main()
{
    string s = "A man, a plan, a canal: Panama";
    Solution a;
    bool result = a.isPalindrome(s);
    std::cout << result << std::endl;
}
